src/CommandingOfficers/Cinder.java

```diff
@@ -69,93 +69,72 @@ public Cinder(GameScenario.GameRules rules)
   }
 
   @Override
-  public void registerForEvents(GameInstance game)
-  {
-    super.registerForEvents(game);
-
-    witchFireListener = new WitchFireListener(this, WitchFireAbility.WITCHFIRE_HP_COST);
-    witchFireListener.registerForEvents(game);
-  }
-  @Override
-  public void unregister(GameInstance game)
+  public void initForGame(GameInstance game)
   {
-    super.unregister(game);
-    witchFireListener.unregister(game);
+    super.initForGame(game);
+    buildCounts = StateTracker.instance(game, BuildCountsTracker.class);
   }
 
   public static CommanderInfo getInfo()
   {
     return coInfo;
   }
 
-  @Override
-  /** Get the list of units this commander can build from the given property type. */
-  public ArrayList<UnitModel> getShoppingList(MapLocation buyLocation)
-  {
-    setPrices(buildCounts.get(buyLocation.getCoordinates()));
-    return super.getShoppingList(buyLocation);
-  }
-
   /*
    * Cinder builds units at 8HP ready to act.
-   * To compensate for the ability to continue producing units in a single turn,
-   * the cost of units increases exponentially for repeated purchases from a single property.
    */
   @Override
   public GameEventQueue receiveCreateUnitEvent(Unit unit)
   {
     XYCoord buildCoords = new XYCoord(unit.x, unit.y);
-    if( this == unit.CO && buildCounts.containsKey(buildCoords) )
+    if( this == unit.CO && myView.isLocationValid(buildCoords) )
```

Comment:
Is there any reason to check `isLocationValid` here? We didn't before, and we don't report anything if the check fails.

Reply:
Yes. Carriers can build Seaplanes, and those Seaplanes aren't on a valid map tile (and thus don't interact with the D2D since IDK what to do with it)

Reply:
Isn't that done via the `UnitProduceEvent`, and thus would never hit this code anyway?

Though it's maybe odd that we have two different events for adding units to the game.

Reply:
Probably yes, and yes.
I suppose it does make some sense - one is adding directly to the board, while the other is adding into another unit's inventory.
Maybe we should just name them better?

Reply:
Yeah, better names might be a good idea.

In the meantime, if we are going to add a check here that could cause the operation to fail, we need to provide some kind of feedback if the operation fails. I'm fine with either printing a warning or removing the unnecessary check.

Reply:
I just looked at the code - both units-producing-units and buildings-producing-units call `receiveCreateUnitEvent()`.
This is a perfectly valid check, and not an error case.

Reply:
Huh. Interesting that `UnitProduceEvent` calls `receiveCreateUnitEvent` instead of having its own listener. Explains why I didn't catch that connection.

Fair enough I guess, but I'm not sure how I feel about having a `UnitProduceEvent` with no listener/re-using an existing listener.

Reply:
IDK, it seems sensible to have an all-in-one place to listen for new units appearing.
Perhaps it also makes sense to provide separate listener hooks for those specific events, though.

Reply:
That's probably what I would have done. That or just make production use the `UnitCreateEvent`. Changing that is certainly out of scope for this PR though.